packages/redux-api-auth-middleware/src/__tests__/refresh-token-middleware.spec.js

```diff
@@ -0,0 +1,132 @@
+import { RSAA, apiMiddleware } from 'redux-api-middleware';
+import configureMockStore from 'redux-mock-store';
+import nodeFetch from 'node-fetch';
+import fetchMock from 'fetch-mock';
+
+import createRefreshTokenMiddleware from '../refresh-token-middleware';
+import createAuthMiddleware from '../auth-middleware';
+import {
+  authReducer,
+  REFRESH_TOKEN_FAILURE,
+  REFRESH_TOKEN_REQUEST,
+  REFRESH_TOKEN_SUCCESS,
+  refreshTokenAction,
+} from '../store';
+import { tokens } from './const';
+
+const refreshEndpoint = 'https://example.com/refresh-token';
+const apiEndpoint = 'https://example.com/users';
+
+const authMiddleware = createAuthMiddleware({
+  header: 'Authorization',
+});
+const refreshTokenMiddleware = createRefreshTokenMiddleware({
+  endpoint: refreshEndpoint,
+  failedAction: { type: 'LOGOUT' },
+});
+const middlewares = [authMiddleware, refreshTokenMiddleware, apiMiddleware];
+const mockStore = configureMockStore(middlewares);
+const flushPromises = () => new Promise(resolve => setImmediate(resolve));
+
+fetchMock.config.overwriteRoutes = true;
+
+describe('Refresh token middleware', () => {
+  let next, store;
+
+  beforeEach(() => {
+    global.fetch = nodeFetch;
+    next = jest.fn();
+
+    store = {
+      dispatch: jest.fn(() => new Promise(resolve => resolve({ error: true }))),
+      getState: () => ({ auth: { ...tokens } }),
+    };
+  });
+
+  afterEach(() => {
+    fetchMock.reset();
+    fetchMock.restore();
+  });
+
+  it('handles only RSAA', () => {
+    const action = {
+      type: 'FOO',
+    };
+    refreshTokenMiddleware(store)(next)(action);
+    expect(next).toHaveBeenCalledWith(action);
+
+    const altAction = refreshTokenAction({ refreshToken: tokens.refreshToken, endpoint: refreshEndpoint });
+    refreshTokenMiddleware(store)(next)(altAction);
+    expect(next).toHaveBeenCalledWith(altAction);
+  });
+
+  it('handles token request reducer', () => {
+    expect(authReducer({}, { type: REFRESH_TOKEN_REQUEST })).toEqual({ isRefreshing: true });
+  });
+
+  it('handles token success reducer', () => {
```

Comment:
missing test cases for SET_TOKEN, CLEAR_TOKEN

Reply:
They are in `auth-token-middleware.spec.js`

Reply:
ok, got confused because its kind of combined scenario (expected reducer action handling and middleware logic test to be separate)